route/route.go

```diff
@@ -232,11 +238,23 @@ func (r *Router) batch(w http.ResponseWriter, req *http.Request) {
 			r.Metrics.IncrementCounter("router_event")
 			ev, err := r.batchedEventToEvent(req, bev)
 			if err != nil {
-				batchedResponses = append(batchedResponses, http.StatusBadRequest)
-				r.Logger.Debugf("event from batch %s failed to convert to event: %s", reqID, err.Error())
+				batchedResponses = append(
+					batchedResponses,
+					&BatchResponse{
+						Status: http.StatusBadRequest,
+						Error:  fmt.Sprintf("failed to convert to event: %s", err.Error()),
+					},
+				)
+				r.Logger.Debugf("event from batch %s failed to process event: %s", reqID, err.Error())
 				continue
 			}
-			batchedResponses = append(batchedResponses, http.StatusAccepted)
+			batchedResponses = append(
+				batchedResponses,
+				&BatchResponse{
+					Status: http.StatusAccepted,
+					Error:  "",
```

Comment:
If you like, you can omit this line and Error will have the default value (which is "")